zerver/views/webhooks/gitlab.py

```diff
@@ -196,6 +228,12 @@ def get_subject_based_on_event(event, payload):
     # type: (str, Dict[str, Any]) -> text_type
     if event == 'Push Hook':
         return u"{} / {}".format(get_repo_name(payload), get_branch_name(payload))
+    elif event == 'Build Hook':
+        return u"{} / {}".format(payload.get('repository').get('name'), get_branch_name(payload))
```

Comment:
Unfortunately not. Gitlab in other hooks has `project` key and there `name` but here we ve got `repository`.